### PR TITLE
chore: rewrite capability to not use sdk.Context

### PR DESCRIPTION
**Pull request by tac0turtle**
## Description

remove sdk.Context from capability and user kvstore service 

when upgrading to 52 we will be able to drop comet as a dep here 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/go.mod [hunk 1]
```diff
@@ -15,7 +15,7 @@ require (
 	cosmossdk.io/math v1.3.0
 	cosmossdk.io/x/upgrade v0.1.4
 	github.com/cometbft/cometbft v0.38.11
-	github.com/cosmos/cosmos-sdk v0.50.9
+	github.com/cosmos/cosmos-sdk v0.50.10-0.20240808075341-156231be8aef
```
[the remark this answers is not in the document]
Comment (tac0turtle) [reply e2e/go.mod:18]: yea need a function, we will tag again to get it, to help unblock this

File: modules/capability/capability_test.go [hunk 2]
```diff
@@ -23,7 +23,11 @@ import (
 	"github.com/cosmos/ibc-go/modules/capability/types"
 )
 
-const mockMemStoreKey = "memory:mock"
+const (
+	mockMemStoreKey   = "memory:mock"
+	bankModuleName    = "bank"
```
[the remark this answers is not in the document]
Comment (tac0turtle) [reply modules/capability/capability_test.go:28]: yea, remove deps for things like this. Makes the code simpler.

File: modules/apps/29-fee/keeper/escrow.go [hunk 3]
```diff
@@ -152,9 +152,10 @@ func (k Keeper) distributePacketFeeOnTimeout(ctx sdk.Context, refundAddr, timeou
 // distributeFee will attempt to distribute the escrowed fee to the receiver address.
 // If the distribution fails for any reason (such as the receiving address being blocked),
 // the state changes will be discarded.
-func (k Keeper) distributeFee(ctx sdk.Context, receiver, refundAccAddress sdk.AccAddress, fee sdk.Coins) {
+func (k Keeper) distributeFee(ctx context.Context, receiver, refundAccAddress sdk.AccAddress, fee sdk.Coins) {
 	// cache context before trying to distribute fees
-	cacheCtx, writeFn := ctx.CacheContext()
+	sdkCtx := sdk.UnwrapSDKContext(ctx) // TODO: remove when Upgrading to 52
```
[the remark this answers is not in the document]
Comment (tac0turtle) [reply modules/apps/29-fee/keeper/escrow.go:157]: https://github.com/cosmos/ibc-go/issues/7223

File: e2e/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (damiannolan) [reply e2e/go.mod:18]: what function was needed that wasn't included in v50.9?

**Review by damiannolan — APPROVED** [review]
Thanks again! 

Just a note, we will be removing capabilities for v10. https://github.com/cosmos/ibc-go/issues/7107

I think its good to just get these changes into main right now tho. We can merge main back into the capability release branch and tag something when needed

**Pull request merged**